pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -275,7 +280,10 @@ protected void handleConnected(CommandConnected connected) {
         }
 
         checkArgument(state == State.SentConnectFrame || state == State.Connecting);
-
+        int maxFrameSize = connected.getMaxMessageSize() - 10 * 1024;
+        NUMBER_OF_MAX_MESSAGE_SIZE.compareAndSet(this, 0, maxFrameSize);
```

Comment:
This compare and set is not needed here. The `CommandConnected` is only received once, so we should just do the replace of the frame decoder.